### ISSUE TITLE
Add missing items

### ISSUE THREAD

**Issue by d3lm** [angular-checklist/angular-checklist#10]
There are a few things that we still need to add. In particular, it's the following items:

- [x] use latest version of everything (General)
- [ ] category for Testing 
- [ ] use cypress for integration and e2e tests (opinionated but we can mark it as optional in the UI, goes into Testing)
- [x] only manipulate the DOM via the Renderer (Components)
- [ ] wrap globals in services
- [ ] use valueChanges over keyup event or a subject (Forms)
- [x] use path mappings to simplify imports (Typescript)

Contributions are more than welcome 🤩If someone wishes to contribute with one of the items above, please create separate pull requests and follow our [contributing guide](https://github.com/angular-checklist/angular-checklist/blob/master/CONTRIBUTING.md).

I also want to point out again that this project is released with a [Code of Conduct](https://github.com/angular-checklist/angular-checklist/blob/master/CODE_OF_CONDUCT.md). By participating in this project you agree to abide by its terms.

**Comment by d3lm** [comment]
Ha, good question. I actually thought about the categories where they belong to but I was struggling so I just put them out without the categories 😆I'll give it another shot and try to come up with categories for each of those items.

**Comment by d3lm** [comment]
For some I tend to have a "General" category but I fear that having such general category will result in a "trash can" where all the items go that cannot be categorized. This is the same I often see in code where people create a shared "models" folder and at the end of the day, people add **everything** to that folder because it's easy first.

**Comment by FUNExtreme** [comment]
Can't say I disagree, however if you decide not to create a general category you'll probably end up with a lot of categories with (potentially) only a single item.
From a contributors standpoint it might scare some people of that there might not yet be a category for their item.
From a maintainers standpoint you could easily propose a new category when a PR is created with an item that shouldn't be in the "general" category.

So I'd dare say the "general" category is the way to go. You'll probably need to keep close track of what's in it though, as new items could warrant a new category in combination with existing items in "general".

**Comment by d3lm** [comment]
Yes agreed. Maybe we can introduce a "General" category. I think it's just our job as contributors to keep close track of where items go. That's why we wanted people to create issues first, so we can discuss these kind of things. So I'd say, let's go with a general one cause it's better to have such rather than having almost empty categories or a whole bunch of categories with just a single item.

**Comment by d3lm** [comment]
So I have added some categories to the items. If you feel like sending PRs for some of the items, please feel free to do that. I am happy to also collaborate and help you out if you need assistance @FUNExtreme. I really appreciate your work and contribution to this project 🙏

**Comment by billyjov** [comment]
@d3lm
 > only manipulate the DOM via the Renderer

I want to write something about this. Should i create the `View` Category ? or add them to existing category ?

**Comment by d3lm** [comment]
Hey @billyjov, cool that you want to contribute!

Regarding the category, I'd say we should add it to `Components` because it already exists and the item would have something to do with components so it's closely related.

**Comment by billyjov** [comment]
@d3lm can you pls update the list of missing items ?

**Comment by KwintenP** [comment]
@billyjov What items are missing atm?

**Comment by billyjov** [comment]
@KwintenP It was about the above list to see which items is on the todo list

**Comment by KwintenP** [comment]
Updated it :)

**Comment by tonyr1971** [comment]
Hi the project is unable to compile, I think we are missing a file:
[0] ERROR in ./src/app/checklist/state/checklist.reducer.ts
[0] Module not found: Error: Can't resolve '../../../assets/content.json' in 'C:\Temp\angular-checklist\src\app\checklist\state'
[0] i ｢wdm｣: Failed to compile.

**Comment by d3lm** [comment]
@tonyr1971 You have to start the build process with `yarn/npm start` which also starts the build process for the content. It seem that you have just started the app and forgot to start the build for the content. Let me know how it goes.
